### ISSUE TITLE
Why is getsensor allowed to return nil?

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#4538]
Say CLT as a thermister, should never return nil from common sense?

**Comment by mck1117** [comment]
nope, if the sensor is failed it returns nil. If you unplug the CLT sensor, what should it return? 0? -1? Those are still valid CLT values. So instead we return nil, which means "no clt value available".

**Comment by ElDominio** [comment]
Most ECUs return a known error value, like 
![image](https://user-images.githubusercontent.com/5685108/188691033-0ad9948e-c3e0-42ed-90f8-935eefdc36ac.png)


The error is logged and a DTC is thrown (to alert everything that CLT is in error), but a value is substituted so things that depend on it go to a safe operating parameter instead of having unhandled errors (like Lua currently is doing)
